x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/SnapshotLifecyclePolicy.java

```diff
@@ -169,6 +171,30 @@ public ActionRequestValidationException validate() {
             }
         }
 
+        if (configuration.containsKey("metadata")) {
+            if (configuration.get("metadata") instanceof Map == false) {
+                err.addValidationError("invalid configuration.metadata [" + configuration.get("metadata") +
```

Comment:
Can we extract `"metadata"` into a field somewhere (and use it in the snapshot request also)? It's always much easier to refactor later if we have to change something if we only have to change a constant rather than a bunch of strings.

Reply:
Done.